File: GestaoAcademicaWeb_old/app/Classes/Endereco.php
<?php
namespace \app\Classes;
/**
 *
 */
 class Endereco{

 private $id_endereco;
 private $logradouro;
 private $numero;
 private $complemento;
 private $bairro;
 private $cidade;
 private $estado;
 private $cep;
 public function __construct($id_endereco = 0,$logradouro= "" ,$numero= "" ,
                             $complemento= "" ,$bairro= "" ,$cidade= "" ,
                             $estado= "" ,$cep= "" ){
 $this->id_endereco = $id_endereco;
 $this->logradouro = $logradouro;
 $this->numero = $numero;
 $this->complemento = $complemento;
 $this->bairro = $bairro;
 $this->cidade = $cidade;
 $this->estado = $estado;
 $this->cep = $cep;

 }

 public static function construct($array){
 $obj = new Endereco();
 $obj->setId_endereco( $array['id_endereco']);
 $obj->setLogradouro( $array['logradouro']);
 $obj->setNumero( $array['numero']);
 $obj->setComplemento( $array['complemento']);
 $obj->setBairro( $array['bairro']);
 $obj->setCidade( $array['cidade']);
 $obj->setEstado( $array['estado']);
 $obj->setCep( $array['cep']);
 return $obj;

 }

 public function getId_endereco(){
 return $this->id_endereco;
 }

 public function setId_endereco($id_endereco){
  $this->id_endereco=$id_endereco;
 }

 public function getLogradouro(){
 return $this->logradouro;
 }

 public function setLogradouro($logradouro){
  $this->logradouro=$logradouro;
 }

 public function getNumero(){
 return $this->numero;
 }

 public function setNumero($numero){
  $this->numero=$numero;
 }

 public function getComplemento(){
 return $this->complemento;
 }

 public function setComplemento($complemento){
  $this->complemento=$complemento;
 }

 public function getBairro(){
 return $this->bairro;
 }

 public function setBairro($bairro){
  $this->bairro=$bairro;
 }

 public function getCidade(){
 return $this->cidade;
 }

 public function setCidade($cidade){
  $this->cidade=$cidade;
 }

 public function getEstado(){
 return $this->estado;
 }

 public function setEstado($estado){
  $this->estado=$estado;
 }

 public function getCep(){
 return $this->cep;
 }

 public function setCep($cep){
  $this->cep=$cep;
 }
 public function equals($object){
 if($object instanceof Endereco){

 if($this->id_endereco!=$object->id_endereco){
 return false;

 }

 if($this->logradouro!=$object->logradouro){
 return false;

 }

 if($this->numero!=$object->numero){
 return false;

 }

 if($this->complemento!=$object->complemento){
 return false;

 }

 if($this->bairro!=$object->bairro){
 return false;

 }

 if($this->cidade!=$object->cidade){
 return false;

 }

 if($this->estado!=$object->estado){
 return false;

 }

 if($this->cep!=$object->cep){
 return false;

 }

 return true;

 }
 else{
 return false;
 }

 }
 public function toString(){

  return "  [id_endereco:" .$this->id_endereco. "]  [logradouro:" .$this->logradouro. "]
            [numero:" .$this->numero. "]  [complemento:" .$this->complemento. "]
            [bairro:" .$this->bairro. "]  [cidade:" .$this->cidade. "]
            [estado:" .$this->estado. "]  [cep:" .$this->cep. "]  " ;
 }

 }


?>

File: GestaoAcademicaWeb/Madodis/src/br/edu/facol/gestaoacademicaweb/pojo/Endereco.java
package br.edu.facol.gestaoacademicaweb.pojo;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.Table;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;

@Entity
@Table(name="TB_ENDERECO")
public class Endereco extends BaseObject {

	@NotNull(message="Logradouro não pode ser nulo ou vazio.")
	@Size(min=1, message="Logradouro não pode ser nulo ou vazio.")
	@Column(name="LOGRADOURO")
	private String logradouro;
	
	@NotNull(message="Numero não pode ser nulo ou vazio.")
	@Size(min=1, message="Numero não pode ser nulo ou vazio.")
	@Column(name="numero")
	private String numero;
	
	@Column(name="complemento")
	private String complemento;
	
	@NotNull(message="CEP não pode ser nulo ou vazio.")
	@Size(min=1, message="CEP não pode ser nulo ou vazio.")
	@Column(name="cep")
	private String cep;
	
	@NotNull(message="Bairro não pode ser nulo ou vazio.")
	@Size(min=1, message="Bairro não pode ser nulo ou vazio.")
	@Column(name="bairro")
	private String bairro;
	
	@NotNull(message="Cidade não pode ser nulo ou vazio.")
	@Size(min=1, message="Cidade não pode ser nulo ou vazio.")
	@Column(name="cidade")
	private String cidade;
	
	@NotNull(message="Estado não pode ser nulo ou vazio.")
	@Column(name="estado")
	@Enumerated(value=EnumType.ORDINAL)
	private Estado estado;
	
	public Endereco() {
		super();
	}

	public String getNumero() {
		return numero;
	}

	public void setNumero(String numero) {
		this.numero = numero;
	}
	

	public String getLogradouro() {
		return logradouro;
	}

	public void setLogradouro(String logradouro) {
		this.logradouro = logradouro;
	}

	public String getComplemento() {
		return complemento;
	}

	public void setComplemento(String complemento) {
		this.complemento = complemento;
	}

	public String getCep() {
		return cep;
	}

	public void setCep(String cep) {
		this.cep = cep;
	}

	public String getBairro() {
		return bairro;
	}

	public void setBairro(String bairro) {
		this.bairro = bairro;
	}

	public String getCidade() {
		return cidade;
	}

	public void setCidade(String cidade) {
		this.cidade = cidade;
	}

	public Estado getEstado() {
		return estado;
	}

	public void setEstado(Estado estado) {
		this.estado = estado;
	}

	@Override
	public String toString() {
		return "Id: " + getId() + ", Logradouro: " + getLogradouro() + ", Numero: " + getNumero();
	}
	

}
